pnc-rest/src/main/java/org/jboss/pnc/rest/endpoint/UserEndpoint.java

```diff
@@ -49,6 +53,7 @@ public UserRest getSpecific(@ApiParam(value = "User id", required = true) @PathP
     @ApiOperation(value = "Creates new User")
     @POST
     public Response createNew(@NotNull @Valid UserRest userRest, @Context UriInfo uriInfo) {
+        validateIdIsNull(userRest);
```

Comment:
Yeah, I think that's a good idea.  It will require changes to the method params in the provider classes, but I think that's ok.  Taking this a step further, do you think it would be good to make the endpoint classes all interfaces, so that they have no functionality and only define the API?